### PR TITLE
fix(chrome-129): show issue page

### PR DESCRIPTION
**Pull request by samdenty**
This makes people aware of https://github.com/stackblitz/bolt.new/issues/86#issuecomment-2395519258

CleanShot 2024-10-07 at 14 53 48@2x

### CODE REVIEW

File: vite.config.ts [hunk 1]
```diff
@@ -24,7 +24,35 @@ export default defineConfig((config) => {
...
+
+        if (raw) {
+          const version = parseInt(raw[2], 10);
+
+          if (version === 129) {
+            res.setHeader('content-type', 'text/html');
+            res.write(
+              '<body><h1>Please use Chrome Canary for testing.</h1><p>Chrome 129 has an issue with JavaScript modules & vite local development, see <a href="https://github.com/stackblitz/bolt.new/issues/86#issuecomment-2395519258">here for more.</a></p><p><b>Note:</b> This only impacts <u>local development</u>. `pnpm run build` and `pnpm run start` will work fine in this browser.</p></body>',
+            );
+            res.end();
```
Comment (Nemikolh) [inline vite.config.ts:48]: Nit: I think this could just be replaced with a single call to `res.end(...)`.

**Review by Nemikolh — DISMISSED** [review]
Looks good :+1: Just a minor comment

**Review by Nemikolh — DISMISSED** [review]

File: vite.config.ts [hunk 2]
```diff
@@ -24,7 +24,34 @@ export default defineConfig((config) => {
...
+      server.middlewares.use((req, res, next) => {
+        const raw = req.headers['user-agent']?.match(/Chrom(e|ium)\/([0-9]+)\./);
+
+        if (raw) {
+          const version = parseInt(raw[2], 10);
+
+          if (version === 129) {
+            res.setHeader('content-type', 'text/html');
+            res.end(
+              '<body><h1>Please use Chrome Canary for testing.</h1><p>Chrome 129 has an issue with JavaScript modules & vite local development, see <a href="https://github.com/stackblitz/bolt.new/issues/86#issuecomment-2395519258">here for more.</a></p><p><b>Note:</b> This only impacts <u>local development</u>. `pnpm run build` and `pnpm run start` will work fine in this browser.</p></body>',
```
Comment (d3lm) [inline vite.config.ts:46]:
```suggestion
              '<body><h1>Please use Chrome Canary for testing.</h1><p>Chrome 129 has an issue with JavaScript modules & Vite local development, see <a href="https://github.com/stackblitz/bolt.new/issues/86#issuecomment-2395519258">for more information.</a></p><p><b>Note:</b> This only impacts <u>local development</u>. `pnpm run build` and `pnpm run start` will work fine in this browser.</p></body>',
```

**Review by d3lm — COMMENTED** [review]
Minor comment.

**Review by Nemikolh — APPROVED** [review]

**Review by Scutua — APPROVED** [review]

**Pull request merged**